pkg/liqoctl/uninstall/preuninstall.go

```diff
@@ -99,15 +116,93 @@ func (o *Options) preUninstall(ctx context.Context) error {
 		}
 	}
 
+	// Search for NamespaceOffloading resources
 	var namespaceOffloadings offloadingv1alpha1.NamespaceOffloadingList
-	if err := o.CRClient.List(ctx, &namespaceOffloadings); errors.IgnoreNoMatchError(err) != nil {
+	if err := errors.IgnoreNoMatchError(o.CRClient.List(ctx, &namespaceOffloadings)); err != nil {
 		return err
 	}
-
 	for i := range namespaceOffloadings.Items {
 		offloading := &namespaceOffloadings.Items[i]
 		errMap.offloading = append(errMap.offloading, offloading.Namespace)
 	}
 
+	// Search for Configuration resources
+	var configurations networkingv1alpha1.ConfigurationList
+	if err := errors.IgnoreNoMatchError(o.CRClient.List(ctx, &configurations)); err != nil {
+		return err
+	}
+	for i := range configurations.Items {
+		addResourceToErrMap(&configurations.Items[i], &errMap, &foreignClusterList)
+	}
+
+	// Search for GatewayServer resources
+	var gatewayServers networkingv1alpha1.GatewayServerList
+	if err := errors.IgnoreNoMatchError(o.CRClient.List(ctx, &gatewayServers)); err != nil {
+		return err
+	}
+	for i := range gatewayServers.Items {
+		addResourceToErrMap(&gatewayServers.Items[i], &errMap, &foreignClusterList)
+	}
+
+	// Search for GatewayClient resources
+	var gatewayClients networkingv1alpha1.GatewayClientList
+	if err := errors.IgnoreNoMatchError(o.CRClient.List(ctx, &gatewayClients)); err != nil {
+		return err
+	}
+	for i := range gatewayClients.Items {
+		addResourceToErrMap(&gatewayClients.Items[i], &errMap, &foreignClusterList)
+	}
+
+	// Search for IP resources
+	var ips ipamv1alpha1.IPList
+	if err := errors.IgnoreNoMatchError(o.CRClient.List(ctx, &ips)); err != nil {
+		return err
+	}
+	for i := range ips.Items {
+		if len(ips.Items[i].GetFinalizers()) > 0 {
+			addResourceToErrMap(&ips.Items[i], &errMap, &foreignClusterList)
+		}
+	}
+
+	// Search for Network resources
+	var networks ipamv1alpha1.NetworkList
+	if err := errors.IgnoreNoMatchError(o.CRClient.List(ctx, &networks)); err != nil {
+		return err
+	}
+	for i := range networks.Items {
+		if len(networks.Items[i].GetFinalizers()) > 0 {
+			addResourceToErrMap(&networks.Items[i], &errMap, &foreignClusterList)
+		}
+	}
+
 	return errMap.getError()
 }
+
+func addResourceToErrMap(obj client.Object, errMap *errorMap, foreignClusters *discoveryv1alpha1.ForeignClusterList) {
+	// Check if object is a networking resource associated with a remote cluster
+	clusterName, found := getRemoteClusterName(obj, foreignClusters)
```

Comment:
I would use the same method to get the name, independently if it is a network resource or not. It can be another resource with the label, or a network resource where the foreignclsuter is no more available